app/src/main/java/org/jellyfin/mobile/fragment/WebViewFragment.kt

```diff
@@ -263,6 +238,52 @@ class WebViewFragment : Fragment(), NativePlayerHost {
         loadUrl(server.hostname)
     }
 
+    private fun showOutdatedWebViewDialog() {
+        AlertDialog.Builder(requireContext()).apply {
+            setTitle(R.string.dialog_web_view_outdated)
+            setMessage(R.string.dialog_web_view_outdated_message)
+            setCancelable(false)
+
+            val webViewPackage = WebViewCompat.getCurrentWebViewPackage(context)
+            if (webViewPackage != null) {
+                val marketUri = Uri.Builder().apply {
+                    scheme("market")
+                    authority("details")
+                    appendQueryParameter("id", webViewPackage.packageName)
+                }.build()
+                val referrerUri = Uri.Builder().apply {
+                    scheme("android-app")
+                    authority(context.packageName)
+                }.build()
+
+                val marketIntent = Intent(Intent.ACTION_VIEW).apply {
+                    data = marketUri
+                    putExtra(Intent.EXTRA_REFERRER, referrerUri)
+                }
+
+                // Only show button if the intent can be resolved
+                if (marketIntent.resolveActivity(context.packageManager) != null) {
+                    setNegativeButton(R.string.dialog_button_check_for_updates) { _, _ ->
+                        startActivity(marketIntent)
+                        requireActivity().finishAfterTransition()
+                    }
+                }
+            }
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+                setPositiveButton(R.string.dialog_button_open_settings) { _, _ ->
+                    startActivity(Intent(Settings.ACTION_WEBVIEW_SETTINGS))
+                    Toast.makeText(context, R.string.toast_reopen_after_change, Toast.LENGTH_LONG).show()
+                    requireActivity().finishAfterTransition()
+                }
+            }
+            setNeutralButton(R.string.dialog_button_ignore) { _, _ ->
+                appPreferences.ignoreWebViewChecks = true
+                // Re-initialize
+                webView.initialize()
```

Comment:
Theoretically a behavior change (as the WebView reference isn't captured but accessed through the getter), but should be fine.